frontend/src/metabase/query_builder/containers/QueryViewer.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useMemo } from "react";
+import _ from "underscore";
+import { useSelector, useDispatch } from "react-redux";
+
+import NativeQueryEditor from "metabase/query_builder/components/NativeQueryEditor";
+import ReadOnlyNotebook from "metabase/query_builder/components/notebook/ReadOnlyNotebook";
+
+import { getMetadata } from "metabase/selectors/metadata";
+import { loadMetadataForCard } from "metabase/questions/actions";
+
+import type { DatasetQuery } from "metabase-types/types/Card";
+
+import Question from "metabase-lib/Question";
+
+export default function QueryViewer({
+  datasetQuery,
+}: {
+  datasetQuery: DatasetQuery;
+}) {
+  const dispatch = useDispatch();
+  const metadata = useSelector(getMetadata, _.isEqual);
+  const card = useMemo(() => ({ dataset_query: datasetQuery }), [datasetQuery]);
+
+  useEffect(() => {
+    async function loadMetadata() {
+      await dispatch(loadMetadataForCard(card as any) as any);
+    }
+    loadMetadata();
+  }, [card, dispatch]);
+
+  const question = new Question(card, metadata);
+
+  const query = question.query();
```

Comment:
I would appreciate any suggestions for improving the use of redux hooks here. We don't really have an established pattern for this yet. 

This works, but certainly isn't very elegant. The `any` typecasting in particular is gross, but our redux actions don't seem to play well with typescript. 😢